### ISSUE TITLE
Technically constrained subordinate CA certificate definition 

### ISSUE THREAD

**Issue by barrini** [cabforum/servercert#492]
In the TLS BRs, we can find the following definition:
Technically Constrained Subordinate CA Certificate: A Subordinate CA certificate which uses a combination of Extended Key Usage **and/or** Name Constraint extensions, as defined within the relevant Certificate Profiles of this document, to limit the scope within which the Subordinate CA Certificate may issue Subscriber or additional Subordinate CA Certificates. (emphasis mine)

That indicates that for a subCA to be technically constrained, the combination could be: EKU, name constraint or EKU+name constraint

In section 7.1.2, we have some specific sub-sections for technically constrained subCAs:
7.1.2.3 for non-TLS subCA (we may review if this sub-section is needed considering that the BRs are now specific for TLS)
7.1.2.4 for precertificate
7.1.2.5 for TLS
All these come with EKU is a MUST and the first two, name constraint is a MAY and in 7.1.2.5 is a MUST.

But, in section 7.1.2.6, which is for non-constrained TLS subCA, the EKU is also a MUST (see 7.1.2.6.1) and the name constraint is a MAY (same as in 7.1.2.3 and 7.1.2.4)

So, according to the definition, basically all subCAs are technically-constrained because of the "or". Thoughts?

I think the easiest way to solve this is by removing the "or" from the definition.

**Comment by barrini** [comment]
And change the MAY with a MUST in sections 7.1.2.3 and 7.1.2.4 if they want to be considered technically constrained

**Comment by aarongable** [comment]
In my opinion, the "and/or" in this definition is a major bug and should be fixed immediately. It is a major bug because it means that any Subordinate CA Certificate which includes an EKU counts as Technically Constrained, and per [Section 8.1](https://github.com/cabforum/servercert/blob/main/docs/BR.md#81-frequency-or-circumstances-of-assessment) therefore _only needs to undergo Self-Assessments_, not full audits. This seems like a significant loophole allowing essentially all current WebPKI Subordinate CA Certificates to be un-audited.

The easiest fix is to change the definition to use the word "and". A slightly larger fix would be to simply remove the reference to EKUs from the definition and say that a CA is Technically Constrained only if it has name-constraints.

-----

In my opinion, the best fix would be to wholly remove the definition of Technically Constrained, and merge all of the Technically Constrained Certificate Profiles into their corresponding non-constrained profiles.

Beyond the auditing difference discussed above, there are two additional carve-outs in the BRs for Technically Constrained certificates:
- OCSP responders only SHOULD NOT respond with "good" for unassigned serials, rather than MUST NOT; and
- CAA checking can be optional, depending on the contract with the Applicant (to whom the technically constrained cert is presumably name-constrained, although the requirements don't actually say that).

I don't think that either of these carve-outs is worth preserving on its own merits. And I personally think that the carve-out allowing certain SubCAs to be un-audited is, despite being based in historical precedent, generally a danger to the WebPKI.

We should survey the current population of name-constrained subordinate CAs that chain up to roots in the WebPKI, then see if any of those were _not_ covered by audits in the past year. If there is no significant population of such certificates then no one is taking advantage of this carve-out and it should be removed posthaste.

**Comment by XolphinMartijn** [comment]
@aarongable I do agree. I would add, we could do both. Remove the "/or" in a quick ballot (or the proposed cleanup ballot), and then separately work on the other items

**Comment by barrini** [comment]
Well, not only the "or" as I indicated in the first post but also to change the MAY with a MUST to avoid inconsistencies in sections 7.1.2.3 and 7.1.2.4. I agree with Martijn that the other questions can be discussed separately

**Comment by timfromdigicert** [comment]
I started a thread for this on the Definitions WG mailing list. I would agree with removing the /or, then working on a better long term fix.

**Comment by aarongable** [comment]
Fair enough. Maybe, if you do a very close reading, there's nothing actually wrong here.

The definition says (emphasis added):
> **Technically Constrained Subordinate CA Certificate**: A Subordinate CA certificate which uses a combination of Extended Key Usage and/or Name Constraint extensions, **as defined within the relevant Certificate Profiles of this document**, to limit the scope within which the Subordinate CA Certificate may issue Subscriber or additional Subordinate CA Certificates.

Those profiles show three things:
- A Technically Constrained Non-TLS Subordinate CA Certificate MUST have an EKU extension, but MUST NOT include id-kp-serverAuth in that extension.
- A Technically Constrained Precertificate Signing CA Certificate MUST have an EKU extension, and MUST only include a specific OID (1.3.6.1.4.1.11129.2.4.4) in that extension.
- A Technically Constrained TLS Subordinate CA Certificate MUST have name constraints

Therefore we can surmise that the underlying definition of Technically Constrained is "either does not have the TLSServerAuth EKU, or if it does, has a name constraint".

I think the definition in Section 1.6.1 should be updated to reflect this. Either it should have all seemingly-normative text removed from it and simply say "A certificate which matches one of the profiles with Technically Constrained in their name"; or it should be more precise and say that the certificate must either have no TLSServerAuth EKU or have a nameConstraint or both.

**Comment by CBonnell** [comment]
> Fair enough. Maybe, if you do a very close reading, there's nothing actually wrong here.

Yes, I think the current language is correct but its presentation can be improved.

> Therefore we can surmise that the underlying definition of Technically Constrained is "either does not have the TLSServerAuth EKU, or if it does, has a name constraint".

Bingo.

> I think the definition in Section 1.6.1 should be updated to reflect this. Either it should have all seemingly-normative text removed from it and simply say "A certificate which matches one of the profiles with Technically Constrained in their name"; or it should be more precise and say that the certificate must either have no TLSServerAuth EKU or have a nameConstraint or both.

I think your second proposed approach is good.

**Comment by CBonnell** [comment]
Also, Chrome root policy already requires third-party audits for technically constrained TLS CAs. Given this, it probably makes sense holistically review the allowances of technically constrained TLS CAs and see if we can entirely dispense with the concept.

**Comment by aarongable** [comment]
Very rough draft proposed language for a short-term definition fix (I still think we should remove this whole concept in the long term):

> **Technically Constrained Subordinate CA Certificate:** A Subordinate CA certificate which either has a Name Constraint extension, or has an Extended Key Usage extension that does not include id-kp-serverAuth or anyExtendedKeyUsage, or both. A Subordinate CA certificate is only considered Technically Constrained if it matches one of the Technically Constrained Profiles in Section 7 of this document.
